### DISCUSSION TITLE
Build steps

### DISCUSSION THREAD

**Discussion by cflaviu** [managarm/managarm#290] [General]
Hi, I could not find build steps. How do you build everything? Then what IDE do you use?

**Comment by avdgrinten** [comment]
As of now, build instructions can be found here: https://github.com/managarm/bootstrap-managarm

I would strongly recommend building with Docker (as described in the link above).

Regarding the IDE: that does not really matter. You can use lightweight editors such as VIM or VSCode or fully blown IDEs (although I do not know how easy it will be to call the build system from an IDE).

**Reply by cflaviu** [reply]
Thank you

**Comment by circutrider21** [comment]
Is it possible to edit managarm code if I build with docker?

**Reply by ArsenArsen** [reply]
yes, container runtimes (including the recommended cbuildrt) only serve to run build commands

**Reply by circutrider21** [reply]
oh, that's nice

**Reply by circutrider21** [reply]
I was trying to work on managarm a while back, but I could never build it with cbuildrt (I was on void Linux, and my default xbps was colliding with managarm's), so I guess I'll try again, but with docker this time 🙂

**Reply by ArsenArsen** [reply]
cbuildrt doesn't use host xbps + we use unmodified xbps, so voids xbps should be fine

**Reply by avdgrinten** [reply]
If cbuildrt doesn't work, that's a bug that we should fix. We *did* fix some bugs in cbuildrt a while ago, so it might just work now.

In general, cbuildrt is quite a bit faster than Docker.

**Reply by circutrider21** [reply]
I'll try building it with docker later this week, if I find the time
